src/matrix/room/timeline/Timeline.js

```diff
@@ -292,7 +296,7 @@ export class Timeline {
     _createEntryFromRelatedEntries(entry, relatedEntry) {
         if (entry.isRedaction) {
             const newEntry = relatedEntry.clone();
-            newEntry.setAsRedacted();
+            newEntry.redact(entry);
```

Comment:
newEntry isn't a NonPersistedEventEntry here when it was fetched from storage (but not in the timeline).